### ISSUE TITLE
Is it possible to add support for Gitlab CI?

### ISSUE THREAD

**Issue by smirnoffmv** [calibreapp/image-actions#257]
I would like to add this application to my project, but as I understand it, it is possible to use it only in Github Actions? Or am I wrong?

**Comment by benschwarz** [comment]
@smirnoffmv Yes, this is possible, although I'm not a Gitlab user so I am not well positioned to advise on writing a howto guide.

At a high level, you would likely need to: 

- Use `image-actions` docker image directly
- Set any required environment variables expected of image actions (e.g.; `GITHUB_` prefixed vars)
- [Make use of the `compressOnly` flag](https://github.com/calibreapp/image-actions?tab=readme-ov-file#run-compression-only)
- Run the command
- Capture the markdown report output
- Post the markdown report on your PR discussion using Gitlab API/CLI/whatever equivalent
- Commit the file changes to your repository/deploy/etc

If you, or someone else is interested in adding this functionality, I'd happily help to review the necessary changes and help get it merged.
